File: iris-did/src/main/java/city/smartb/iris/did/model/DIDService.java
package city.smartb.iris.did.model;

import city.smartb.iris.jsonld.JsonLdObject;
import city.smartb.iris.jsonld.reader.JsonFieldReader;

import java.util.Map;

public class DIDService extends JsonLdObject {

	public static final String JSON_LD_SERVICE_ENDPOINT = "serviceEndpoint";

	public DIDService(Map<String, Object> json) {
		super(json);
	}

	public DIDService(Map<String, Object> json, JsonFieldReader reader) {
		super(json, reader);
	}

	public String getServiceEndpoint() {
		return this.get(JSON_LD_SERVICE_ENDPOINT).asString();
	}

}

File: iris-ld/iris-did/src/main/kotlin/city/smartb/iris/ld/did/DIDService.kt
package city.smartb.iris.ld.did

import city.smartb.iris.ld.jsonld.JsonLdObject
import city.smartb.iris.ld.jsonld.reader.JsonFieldReader

class DIDService : JsonLdObject {
    constructor(json: Map<String, Any>) : super(json)
    constructor(json: Map<String, Any>, reader: JsonFieldReader) : super(json, reader)

    val serviceEndpoint: String?
        get() = this[JSON_LD_SERVICE_ENDPOINT].asString()

    companion object {
        const val JSON_LD_SERVICE_ENDPOINT = "serviceEndpoint"
    }
}
